src/routing_node.rs

```diff
@@ -1039,8 +1039,12 @@ impl<F> RoutingNode<F> where F: Interface {
             return true;
         }
 
-        let close_group = self.routing_table.our_close_group();
-        closer_to_target(&address, &self.routing_table.our_close_group().pop().unwrap().id(), &self.own_name)
+        match self.routing_table.our_close_group().pop() {
+            Some(furthest_close_node) => {
+                closer_to_target_or_equal(&address, &furthest_close_node.id(), &self.own_name)
```

Comment:
Where what is happening? Turns out the' not for us' issue is a non issue just now anyway. The 'or_equal' is including the furthest close node as part of the group which both Ben and I think is correct.